pkg/api/defaults-custom-cloud-profile.go

```diff
@@ -130,6 +143,8 @@ func (p *Properties) SetAzureStackCloudSpec() error {
 			//KubernetesSpecConfig
 			asccKubernetesSpecConfig := ascc.KubernetesSpecConfig
 			azsKubernetesSpecConfig := azureStackCloudSpec.KubernetesSpecConfig
+
+			azureStackCloudSpec.KubernetesSpecConfig.AzureTelemetryPID = helpers.EnsureString(asccKubernetesSpecConfig.AzureTelemetryPID, DefaultAzurestackDeployTelemetryPID)
```

Comment:
The reason we don't bother to set a scale- or upgrade-specific ID here is because the `AzureEnvironmentSpecConfig` will never be non-nil in a scale or upgrade scenario?

Reply:
@jackfrancis that is exactly right. scale and upgrade scenarios are covered above.